### PR TITLE
feature(loggin): log app events to Application Insight

### LINKED ISSUE #325: Add logging
We must be able to track some simple usage of the app.

Suggest shipping logs from API to Azure AppAnalytics.
Could also look into some frontend alternatives for tracking.

### PR DESCRIPTION
**Pull request by soofstad**
## Why is this pull request needed?

- Track usage of app

## What does this pull request change?

- Adds Azure App insight log exporting in API (Most of this is stolen from NeqSimAPI)
- Bicep file for creating Azure resources
- Adds eventTracking in SPA. Tracking data is sent to same Application Insight instance as API logs

NOTE: Add github action secrets before merging

## Issues related to this change:
closes #325

### CODE REVIEW

File: .github/workflows/on-push-main-branch.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
       oauth-redirect-url: "https://proxy-mercury-dev.radix.equinor.com"
     secrets:
       LIBHG_REPO_DEPLOY_KEY: ${{ secrets.LIBHG_REPO_DEPLOY_KEY }}
+      APP_INSIGHT_CON_STRING: ${{ secrets.APP_INSIGHT_CON_STRING_TEST }}
```
Comment (ingeridhellen) [inline .github/workflows/on-push-main-branch.yaml:36]: Shouldn't this be dev instead of test?

**Review by ingeridhellen — APPROVED** [review]

File: IaC/main.bicep [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@description('Specifies the location for resources.')
+param location string = 'norwayeast'
+@allowed(['dev', 'test', 'prod'])
```
[the remark this answers is not in the document]
Comment (soofstad) [reply IaC/main.bicep:3]: Just different allowed environments. No definition for each here.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -49,28 +49,28 @@ repos:
...
-    rev: 5.10.1
+    rev: 5.12.0
     hooks:
       - id: isort
         name: isort (python)
         files: ^api/.*\.py$
 
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v2.7.1
+    rev: v3.0.0-alpha.4
```
[the remark this answers is not in the document]
Comment (soofstad) [reply .pre-commit-config.yaml:65]: reverted

File: .github/workflows/on-push-main-branch.yaml (hunk 1 above)
Comment (soofstad) [reply .github/workflows/on-push-main-branch.yaml:36]: fixed

**Pull request merged**